lib/drivers/bayerContourNext.js

```diff
@@ -271,110 +272,134 @@ module.exports = function (config) {
     if(markers.indexOf('C') !== -1) {
       debug('Marking as control test');
       return true;
-    }else{
+    } else {
       return false;
     }
   };
 
-  var getOneRecord = function (record, data, callback) {
-    parseDataRecord(record,function(err,r){
-      var robj = {};
-      if (err) {
-          debug('Failure trying to read record', record);
-          debug(err);
-          return callback(err, null);
+  var getOneRecord = function (data, callback) {
+    var cmd = buildAckPacket();
+    bcnCommandResponse(cmd, function (err, record) {
+      // TODO - add error handling
```

Comment:
According to the Contour ASTM protocol doc, if a sequence number is out of order, or the checksum doesn't match at this point, we should send a NAK. At the moment, we're always just sending an ACK, and not caring about the sequence number. This doesn't seem like too much of a hassle with USB devices, but could be a problem later if we want to use this code for serial devices (where interference and retransmits are more of a factor).